velox/exec/Exchange.h

```diff
@@ -209,20 +209,26 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   using Factory = std::function<std::shared_ptr<ExchangeSource>(
       const std::string& taskId,
       int destination,
-      std::shared_ptr<ExchangeQueue> queue)>;
+      std::shared_ptr<ExchangeQueue> queue,
+      memory::MemoryPool* pool)>;
 
   ExchangeSource(
       const std::string& taskId,
       int destination,
-      std::shared_ptr<ExchangeQueue> queue)
-      : taskId_(taskId), destination_(destination), queue_(std::move(queue)) {}
+      std::shared_ptr<ExchangeQueue> queue,
+      memory::MemoryPool* pool = nullptr)
```

Comment:
Let's not provide a default. It is hard to reason about parameters with default values and it is also hard to make sure the code paths work with null and non-null pool.